src/tool/hpcrun/fnbounds/fnbounds_client.c

```diff
@@ -484,17 +501,53 @@ launch_server(void)
   }
 
   // For safety, we don't assume the direction of stack growth
-  pid = auditor_exports->clone(hpcfnbounds_child,
-    &server_stack[SERVER_STACK_SIZE * 1024], SIGCHLD, &fds);
+  child_pid = auditor_exports->clone(hpcfnbounds_child,
+    &server_stack[SERVER_STACK_SIZE * 1024], SIGCHLD | CLONE_VM, &fds);
 
-  if (pid < 0) {
+  if (child_pid < 0) {
```

Comment:
hpcfnbounds_child always returns 0, which would mean that child_pid is always 0 and the server launch is never considered as a failure. I think this should be checking fds.pid instead, which is where hpcfnbounds_child actually returns the pid of the grandchild.

Reply:
This is checking whether the clone for the child shim failed, `clone()` returns the pid regardless of the process body's return value. Line 543 down lower checks `fds.pid` after the child shim has successfully completed.